cmd/dgraph/main.go

```diff
@@ -519,6 +517,11 @@ func setupServer() {
 	go serveHTTP(httpl)
 	go serveHTTP(http2)
 
+	go func() {
+		<-closeCh
+		l.Close()
```

Comment:
Since the TCPListener doesn't close already accepted connections https://golang.org/pkg/net/#TCPListener.Close. The present request would return successfully. Maybe add that as a comment.